packages/server/README.md

```diff
@@ -1,226 +1,305 @@
+// Sets a global logger 
+OpenFeature.setLogger(logger);
 
-```typescript
-// implement logger
-class MyLogger implements Logger {
-  error(...args: unknown[]): void {
-    // implement me
-  }
-  warn(...args: unknown[]): void {
-    // implement me
-  }
-  info(...args: unknown[]): void {
-    // implement me
-  }
-  debug(...args: unknown[]): void {
-    // implement me
-  }
-}
-
-// set the logger
-OpenFeature.setLogger(new MyLogger());
+// Sets a client logger
+const client = OpenFeature.getClient();
+client.setLogger(logger);
 ```
 
-### Named clients:
-
-You can have several clients, that can be referenced by a name.
-Every client can have a different provider assigned. If no provider is assigned to a named client, the global default
-provider is used.
-
-```typescript
-import { OpenFeature, ProviderEvents } from '@openfeature/web-sdk';
+### Named clients
 
-OpenFeature.setProvider(new YourProviderOfChoice())
-OpenFeature.setProvider("client-1", new YourOtherProviderOfChoice())
+Clients can be given a name.
+A name is a logical identifier which can be used to associate clients with a particular provider.
+If a name has no associated provider, the global provider is used.
 
-// Uses YourProviderOfChoice (the default)
-const unnamedClient = OpenFeature.getClient()
+```ts
+import { OpenFeature } from "@openfeature/js-sdk";
 
-// Uses YourOtherProviderOfChoice as it is set explicitly
-const client1 = OpenFeature.getClient("client-1")
+// Registering the default provider
+OpenFeature.setProvider(NewLocalProvider());
+// Registering a named provider
+OpenFeature.setProvider("clientForCache", new NewCachedProvider());
 
-// Uses YourProviderOfChoice as no provider is set
-const client2 = OpenFeature.getClient("client-2")
+// A Client backed by default provider
+const clientWithDefault = OpenFeature.getClient();
+// A Client backed by NewCachedProvider
+const clientForCache = OpenFeature.getClient("clientForCache");
 ```
 
-### Events:
+### Eventing
 
-Events provide a way to react to state changes in the provider or underlying flag management system.
-You can listen to events of either the OpenFeature API or individual clients.
+Events allow you to react to state changes in the provider or underlying flag management system, such as flag definition changes, provider readiness, or error conditions.
+Initialization events (`PROVIDER_READY` on success, `PROVIDER_ERROR` on failure) are dispatched for every provider.
+Some providers support additional events, such as `PROVIDER_CONFIGURATION_CHANGED`.
 
-The events after initialization, `PROVIDER_READY` on success, `PROVIDER_ERROR` on failure during initialization,
-are dispatched for every provider.
-However, other event types may not be supported by your provider.
 Please refer to the documentation of the provider you're using to see what events are supported.
```

Comment:
Should we maybe add a link to the spec for this?
This e.g. https://openfeature.dev/specification/types#provider-events